From da1ae5b54fed6f663363c826a8a01b63a2ac52fc Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Sat, 16 Dec 2023 12:47:50 -0500
Subject: [PATCH 02/43] init: Update set_low_mem function

---
 init.sh | 18 +++++++++++++-----
 1 file changed, 13 insertions(+), 5 deletions(-)

diff --git a/init.sh b/init.sh
index 09edac8..cce3ee1 100644
--- a/init.sh
+++ b/init.sh
@@ -727,17 +727,25 @@ function init_cpu_governor()
 
 function set_lowmem()
 {
-	# 512 MB size in kB : https://source.android.com/devices/tech/perf/low-ram
-	SIZE_512MB=2048000
+	# 3GB size in kB : https://source.android.com/devices/tech/perf/low-ram
+	SIZE_3GB=3145728
 
 	mem_size=`cat /proc/meminfo | grep MemTotal | tr -s ' ' | cut -d ' ' -f 2`
 
-	if [ "$mem_size" -le "$SIZE_512MB" ]
+	if [ "$mem_size" -le "$SIZE_3GB" ]
 	then
-		setprop ro.config.low_ram true
+		setprop ro.config.low_ram ${FORCE_LOW_MEM:-true}
 	else
-		setprop ro.config.low_ram false
+		# Choose between low-memory vs high-performance device. 
+		# Default = false.
+		setprop ro.config.low_ram ${FORCE_LOW_MEM:-false}
 	fi
+
+	# Use free memory and file cache thresholds for making decisions 
+	# when to kill. This mode works the same way kernel lowmemorykiller 
+	# driver used to work. AOSP Default = false, Our default = true
+	setprop ro.lmk.use_minfree_levels ${FORCE_MINFREE_LEVELS:-true}
+	
 }
 
 function set_custom_ota()
-- 
2.34.1

