From b79becf67ba437985e2f80bcdebfba6beb6d3e31 Mon Sep 17 00:00:00 2001
From: electrikjesus <electrikjesus@gmail.com>
Date: Thu, 7 Mar 2024 09:24:17 -0500
Subject: [PATCH 11/43] bass: updates packages.mk

---
 packages.mk | 40 ++++++++++++++++++++++++++++++++++++++++
 1 file changed, 40 insertions(+)

diff --git a/packages.mk b/packages.mk
index efcee7a..e3a6db7 100644
--- a/packages.mk
+++ b/packages.mk
@@ -166,3 +166,43 @@ PRODUCT_PACKAGES += iptsd_runner iptsd \
                     iptsd-dump \
                     iptsd-perf
 endif
+
+# SeedVault
+PRODUCT_PACKAGES += \
+    Seedvault \
+    BasicDreams \
+    
+# Include dhcptool
+PRODUCT_PACKAGES += \
+    libnetutils \
+    dhcpdbg
+
+# Bliss Ethernet Manager
+ifeq ($(USE_BLISS_ETHERNET_MANAGER), true)
+PRODUCT_PACKAGES += \
+    BlissEthernetManagerApp
+
+endif
+
+# Bliss Power Manager
+ifeq ($(USE_BLISS_POWER_MANAGER), true)
+PRODUCT_PACKAGES += \
+    BlissPowerManagerApp
+
+endif
+
+# Bliss Kiosk App
+ifeq ($(USE_BLISS_KIOSK_LAUNCHER), true)
+
+PRODUCT_PACKAGES += \
+    BlissKioskLauncher
+
+endif
+
+# Bliss Restricted Launcher
+ifeq ($(USE_BLISS_RESTRICTED_LAUNCHER), true)
+
+PRODUCT_PACKAGES += \
+    BlissRestrictedLauncher
+
+endif
-- 
2.34.1

