From 7266636fb49647650087f407ce1771288e95d0eb Mon Sep 17 00:00:00 2001
From: electrikjesus <electrikjesus@gmail.com>
Date: Thu, 7 Mar 2024 09:25:47 -0500
Subject: [PATCH 13/43] bass: treble updates for cameras

---
 treble.mk | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/treble.mk b/treble.mk
index aba5094..ef0a0c4 100644
--- a/treble.mk
+++ b/treble.mk
@@ -31,8 +31,22 @@ PRODUCT_PACKAGES += \
     android.hardware.bluetooth@1.1-service.btlinux \
     android.hardware.bluetooth.audio@2.1-impl \
 
+# Camera Device
+PRODUCT_PACKAGES += \
+    android.hardware.camera.common@1.0 \
+    android.hardware.camera.device@1.0 \
+    android.hardware.camera.device@3.2 \
+    android.hardware.camera.device@3.4 \
+    android.hardware.camera.device@3.5
+
 # Camera HAL
 PRODUCT_PACKAGES += \
+    android.hardware.camera.provider@2.4-external-service \
+	android.hardware.camera.provider@2.4-service_64 \
+	android.hardware.camera.provider@2.4-impl \
+    android.hardware.camera.provider@2.6 \
+    android.hardware.camera.provider@2.6-impl \
+    android.hardware.camera.provider@2.6-service_64 \
     camera.x86
 
 # Media codec
-- 
2.34.1

