From d7d401e413de17e4a2c6ace637aa75f83c3cef98 Mon Sep 17 00:00:00 2001
From: electrikjesus <electrikjesus@gmail.com>
Date: Tue, 2 Jan 2024 21:07:37 -0500
Subject: [PATCH 17/43] Move mcime to it's own config in init

---
 init.sh | 20 +++++++++++++++-----
 1 file changed, 15 insertions(+), 5 deletions(-)

diff --git a/init.sh b/init.sh
index 9d47ed8..1692732 100644
--- a/init.sh
+++ b/init.sh
@@ -830,6 +830,20 @@ function set_custom_package_perms()
 		am start -a android.intent.action.MAIN -c android.intent.category.HOME
 	fi
 
+	# MultiClientIME
+	exists_mcime=$(pm list packages com.example.android.multiclientinputmethod | grep -c com.example.android.multiclientinputmethod)
+	if [ $exists_mcime -eq 1 ]; then
+		# Enable desktop mode on external display (required for MultiDisplay Input)
+		settings put global force_desktop_mode_on_external_displays 1
+	fi
+
+	# ZQYMultiClientIME
+	exists_zqymcime=$(pm list packages com.zqy.multidisplayinput | grep -c com.zqy.multidisplayinput)
+	if [ $exists_zqymcime -eq 1 ]; then
+		# Enable desktop mode on external display (required for MultiDisplay Input)
+		settings put global force_desktop_mode_on_external_displays 1
+	fi
+
 	# BlissRestrictedLauncher
 	exists_restlauncher=$(pm list packages com.bliss.restrictedlauncher | grep -c com.bliss.restrictedlauncher)
 	if [ $exists_restlauncher -eq 1 ]; then
@@ -843,7 +857,7 @@ function set_custom_package_perms()
 			chmod 664 /data/misc/rlconfig/admin
 		fi
 		set_property persist.wm.debug.hide_navbar_window 1
-		
+				
 		pm grant com.bliss.restrictedlauncher android.permission.SYSTEM_ALERT_WINDOW
 		pm set-home-activity "com.bliss.restrictedlauncher/.activities.LauncherActivity"
 		am start -a android.intent.action.MAIN -c android.intent.category.HOME
@@ -924,10 +938,6 @@ function set_custom_package_perms()
 		# allow displaying over other apps if in Go mode
 		settings put system alert_window_bypass_low_ram 1
 
-		# Enable desktop mode on external display (required for MultiDisplay Input)
-		settings put global force_desktop_mode_on_external_displays 1
-		settings put global force_allow_on_external 1
-
 		# Only if PC_MODE is 1
 		if [ $PC_MODE -eq 1 ]; then
 			
-- 
2.34.1

