From f6d396ec729d94fdd84a00fd6ed59b381ac3643c Mon Sep 17 00:00:00 2001
From: electrikjesus <electrikjesus@gmail.com>
Date: Thu, 11 Jan 2024 18:11:55 -0500
Subject: [PATCH 20/43] Remove hide_navbar_window from configs

---
 init.sh | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/init.sh b/init.sh
index bc7dda3..771b7d9 100644
--- a/init.sh
+++ b/init.sh
@@ -856,7 +856,6 @@ function set_custom_package_perms()
 			chmod 775 /data/misc/rlconfig
 			chmod 664 /data/misc/rlconfig/admin
 		fi
-		set_property persist.wm.debug.hide_navbar_window 1
 				
 		pm grant com.bliss.restrictedlauncher android.permission.SYSTEM_ALERT_WINDOW
 		pm set-home-activity "com.bliss.restrictedlauncher/.activities.LauncherActivity"
@@ -984,9 +983,6 @@ function set_custom_package_perms()
 		    	settings put global development_settings_enabled 1
 			fi
 
-			# Disable Launcher3 Taskbar
-			set_property persist.wm.debug.hide_navbar_window ${HIDE_NAVBAR_WINDOW:-1}
-
 			[ -n "$SET_SMARTDOCK_DEFAULT" ] && pm set-home-activity "cu.axel.smartdock/.activities.LauncherActivity" || pm set-home-activity "com.android.launcher3/.LauncherProvider"
 			
 		fi
-- 
2.34.1

