From ad1fde0a04d5e5c5d532b8bf4ba0177556c9c853 Mon Sep 17 00:00:00 2001
From: electrikjesus <electrikjesus@gmail.com>
Date: Tue, 2 Jan 2024 21:27:04 -0500
Subject: [PATCH 13/13] Use BLISS_BUILD_ZIP if defined for .iso filename

---
 Android.mk | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Android.mk b/Android.mk
index 711b3f47..1643c726 100644
--- a/Android.mk
+++ b/Android.mk
@@ -185,7 +185,11 @@ ROM_VENDOR_VERSION := $(RELEASE_OS_TITLE)-$(TARGET_ARCH)-$(shell date +%Y%m%d%H%
 endif
 
 BUILD_NAME_VARIANT := $(ROM_VENDOR_VERSION)
+ifeq ($(BLISS_BUILD_ZIP),)
 ROM_VENDOR_VERSION := $(RELEASE_OS_TITLE)$(BLISS_SPECIAL_VARIANT)-$(TARGET_ARCH)-$(shell date +%Y%m%d%H)
+else
+ROM_VENDOR_VERSION := $(BLISS_BUILD_ZIP)
+endif
 
 ISO_IMAGE := $(PRODUCT_OUT)/$(ROM_VENDOR_VERSION).iso
 $(ISO_IMAGE): $(boot_dir) $(BUILT_IMG)
-- 
2.34.1

