From e60c53becf48ab5304c9bec5d1d6949a3a16603b Mon Sep 17 00:00:00 2001
From: electrikjesus <electrikjesus@gmail.com>
Date: Mon, 23 Oct 2023 12:45:43 -0400
Subject: [PATCH 13/19] installer: reword for A/B updates

---
 install/scripts/1-install | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/install/scripts/1-install b/install/scripts/1-install
index 431d7af0..5070990c 100644
--- a/install/scripts/1-install
+++ b/install/scripts/1-install
@@ -956,7 +956,7 @@ if you already have a boot solution in place."
 			[ -d $f ] || chmod 644 $f
 		done
 		
-		dialog --infobox "\n Preparing install for OTA updates..." 5 42
+		dialog --infobox "\n Preparing install for A/B updates..." 5 42
 
 		if [ "$skip_ota_prepare" -eq 0 ]; then
 			dialog --backtitle "$BACK_TITLE" --infobox "\n Preparing install for OTA updates..." 6 41
-- 
2.34.1

