From 54f044f39856071e5bef578cebf7f3d22698e0ac Mon Sep 17 00:00:00 2001
From: electrikjesus <electrikjesus@gmail.com>
Date: Wed, 31 Jan 2024 11:05:20 -0500
Subject: [PATCH 19/19] Update EFI entry creation to make sure things are
 created properly

---
 install/scripts/1-install | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/install/scripts/1-install b/install/scripts/1-install
index 4f636797..1b11863e 100644
--- a/install/scripts/1-install
+++ b/install/scripts/1-install
@@ -587,7 +587,13 @@ install_to()
 			adialog --title " Question " --yesno "\nEFI boot entries for previous Android-x86 installations were found.\n\nDo you wish to delete them?" 10 61
 			[ $? -eq 0 ] && while read entry; do efibootmgr -Bb "$entry" > /dev/tty4 2>&1; done < /tmp/efientries
 		fi
-		efibootmgr -v -c -d /dev/$disk -p $esp -L "Android-x86 $VER" -l $efidir/$bootefi > /dev/tty4 2>&1
+		# Check if the entry has been created, if not, it may due to the efivar got full
+		# Remove all /sys/firmware/efi/efivars/dump-* and run the command again
+		efibootmgr -u -v -c -d /dev/$disk -p $esp -L "Android-x86 $VER" -l $efidir/$bootefi > /dev/tty4 2>&1
+		if [ $? -eq 1 ] || [ ! $(efibootmgr | grep "Android-x86 $VER") ]; then 
+			rm -rf /sys/firmware/efi/efivars/dump-*
+			efibootmgr -u -v -c -d /dev/$disk -p $esp -L "Android-x86 $VER" -l $efidir/$bootefi > /dev/tty4 2>&1
+		fi
 
 		if [ -s efi/startup.nsh ]; then
 			sed -i "s|\\\\efi\\\\Android|$efidir|; s|/|\\\\|g" efi/startup.nsh
-- 
2.34.1

