From 490523a17b5501a80bd54b1c5eb5e147c05b8e42 Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Thu, 15 Jun 2023 15:50:08 -0400
Subject: [PATCH 01/20] Do not build bootanimation

---
 config/common.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config/common.mk b/config/common.mk
index 377250a9..7a1cb390 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -172,7 +172,7 @@ PRODUCT_COPY_FILES += \
     vendor/bliss/config/permissions/privapp-permissions-omadm.xml:$(TARGET_COPY_OUT_PRODUCT)/etc/permissions/privapp-permissions-omadm.xml
 
 # Bliss Bootanimation
--include vendor/bliss/config/bootanimation.mk
+# -include vendor/bliss/config/bootanimation.mk
 
 # Bliss Versioning System
 -include vendor/bliss/config/versions.mk
-- 
2.34.1

