From 17dc0dc7faa0708963bf768eb4bae65fcd7c87ff Mon Sep 17 00:00:00 2001
From: electrikjesus <electrikjesus@gmail.com>
Date: Fri, 3 Nov 2023 10:45:04 -0400
Subject: [PATCH 04/20] Update BLISS_MINIMAL_PACKAGES

---
 config/bliss_packages.mk | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/config/bliss_packages.mk b/config/bliss_packages.mk
index 8ef9db21..1c299a2c 100644
--- a/config/bliss_packages.mk
+++ b/config/bliss_packages.mk
@@ -28,17 +28,23 @@ endif
 
 else
 
+ifneq ($(BLISS_MINIMAL_PACKAGES), true)
 PRODUCT_PACKAGES += \
     Contacts \
     Dialer \
     SimpleGalleryPro
 
-# END Minimal added packages
-endif
-
 # Bliss Packages
 PRODUCT_PACKAGES += \
     DeskClock \
+    WallpaperPicker2
+
+endif # BLISS_MINIMAL_PACKAGES
+
+endif # BLISS_SUPER_VANILLA
+
+# Bliss Packages
+PRODUCT_PACKAGES += \
     LatinIME \
     Recorder \
     LineageParts \
@@ -46,7 +52,6 @@ PRODUCT_PACKAGES += \
     Profiles \
     BlissUpdater \
     Seedvault \
-    WallpaperPicker2 \
     OmniStyle \
     GameSpace \
     OmniJaws
-- 
2.34.1

