From ceff0647d0e94863ac19029c3c02e8c41661bc61 Mon Sep 17 00:00:00 2001
From: electrikjesus <electrikjesus@gmail.com>
Date: Mon, 6 Nov 2023 18:35:32 -0500
Subject: [PATCH 05/20] Add SystemUI to dexpreopt speed apps

---
 config/common.mk | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/config/common.mk b/config/common.mk
index 7a1cb390..6b2f089e 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -186,6 +186,10 @@ PRODUCT_COPY_FILES += \
 # Bliss Overlays
 -include vendor/overlays/bliss_overlays.mk
 
+# Dex preopt
+PRODUCT_DEXPREOPT_SPEED_APPS += \
+    SystemUI
+
 # Gapps
 ifeq ($(BLISS_BUILD_VARIANT), gapps)
 $(call inherit-product, vendor/gapps/x86_64/x86_64-vendor.mk)
-- 
2.34.1

