From c14a4c621fff3071bc98acc8f53616107ebf8048 Mon Sep 17 00:00:00 2001
From: electrikjesus <electrikjesus@gmail.com>
Date: Mon, 6 Nov 2023 18:35:55 -0500
Subject: [PATCH 06/20] Do not enable FaceUnlock by default

---
 config/common.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config/common.mk b/config/common.mk
index 6b2f089e..12388c5b 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -153,7 +153,7 @@ PRODUCT_PACKAGES += \
     StitchImage
 
 # Face Unlock
-TARGET_FACE_UNLOCK_SUPPORTED ?= true
+# TARGET_FACE_UNLOCK_SUPPORTED ?= true
 ifeq ($(TARGET_FACE_UNLOCK_SUPPORTED),true)
 PRODUCT_PACKAGES += \
     FaceUnlockService
-- 
2.34.1

