From 3b89c3565194db2bd66ef704d4161d6b9a88b623 Mon Sep 17 00:00:00 2001
From: electrikjesus <electrikjesus@gmail.com>
Date: Sat, 30 Dec 2023 19:55:00 -0500
Subject: [PATCH 08/11] bass: Use RELEASE_OS_TITLE

---
 config/versions.mk | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/config/versions.mk b/config/versions.mk
index 59373df3..9d5ed213 100644
--- a/config/versions.mk
+++ b/config/versions.mk
@@ -39,7 +39,14 @@ else
     BLISS_BUILDTYPE := UNOFFICIAL
 endif
 
-BLISS_TYPE_NAME := Bliss$(BLISS_SPECIAL_VARIANT)
+# See if RELEASE_OS_TITLE is defined, and if not, use the current title
+ifeq ($(RELEASE_OS_TITLE),)
+RELEASE_OS_TITLE := Bliss
+else
+RELEASE_OS_TITLE := $(RELEASE_OS_TITLE)
+endif
+
+BLISS_TYPE_NAME := $(RELEASE_OS_TITLE)$(BLISS_SPECIAL_VARIANT)
 
 # Set BLISS version
 ifdef BLISS_RELEASE
-- 
2.34.1

