From a56693e42a6a592102cd9134a6b1b7d23a3f2499 Mon Sep 17 00:00:00 2001
From: electrikjesus <electrikjesus@gmail.com>
Date: Sun, 10 Dec 2023 11:50:54 -0500
Subject: [PATCH 10/20] Include navigation bar overlays

---
 config/bliss_packages.mk | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/config/bliss_packages.mk b/config/bliss_packages.mk
index 1c299a2c..c83ec404 100644
--- a/config/bliss_packages.mk
+++ b/config/bliss_packages.mk
@@ -116,3 +116,9 @@ PRODUCT_PACKAGES += \
     LiveWallpapersPicker \
     VisualizationWallpapers \
     librs_jni
+
+# Customizations
+PRODUCT_PACKAGES += \
+    LineageNavigationBarNoHint \
+    NavigationBarMode2ButtonOverlay \
+    NavigationBarMode3ButtonOverlay
-- 
2.34.1

