From 976a4d55831c99e54a16e4f33bd35c708de03284 Mon Sep 17 00:00:00 2001
From: electrikjesus <electrikjesus@gmail.com>
Date: Sat, 30 Dec 2023 19:52:45 -0500
Subject: [PATCH 15/20] Update .gitignore

---
 .gitignore | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/.gitignore b/.gitignore
index 931ce07a..1381e1bc 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,2 +1,4 @@
 proprietary/*
 prebuilt/archive
+blissethernet/*
+config/watermark/*
\ No newline at end of file
-- 
2.34.1

