From a93c774c0a189c8d1aded6b0d5fe254959992ff5 Mon Sep 17 00:00:00 2001
From: electrikjesus <electrikjesus@gmail.com>
Date: Wed, 6 Mar 2024 15:55:13 -0500
Subject: [PATCH 18/20] Allow using minimal FOSS apps on vanilla builds

---
 config/common.mk | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/config/common.mk b/config/common.mk
index 0402f5fe..a0f78ce3 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -204,6 +204,12 @@ endif
 ifeq ($(BLISS_BUILD_VARIANT), foss)
 $(call inherit-product, vendor/foss/foss.mk)
 endif
+# We also make a minimal FOSS apps variant that only includes Neo-Store
+ifeq ($(BLISS_BUILD_VARIANT), vanilla)
+ifeq ("$(USE_FOSSAPPS)","true")
+$(call inherit-product, vendor/foss/foss.mk)
+endif
+endif
 
 # Go-gapps
 ifeq ($(BLISS_BUILD_VARIANT), goapps)
-- 
2.34.1

