From ced7f7499133ad83053608ad42c4219f7150a604 Mon Sep 17 00:00:00 2001
From: electrikjesus <electrikjesus@gmail.com>
Date: Wed, 6 Mar 2024 17:55:41 -0500
Subject: [PATCH 20/20] ignore signature folder

---
 .gitignore | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/.gitignore b/.gitignore
index 1381e1bc..1188a8e0 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,4 +1,5 @@
 proprietary/*
 prebuilt/archive
 blissethernet/*
-config/watermark/*
\ No newline at end of file
+config/watermark/*
+config/signing/*
\ No newline at end of file
-- 
2.34.1

