From cd79c73d462b60c7fd0bc43a0e344a28ec00de9f Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Sat, 16 Dec 2023 12:45:33 -0500
Subject: [PATCH 01/43] init: Use early suspend for taifa elimu-tab

---
 init.sh | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/init.sh b/init.sh
index b99eab8..09edac8 100644
--- a/init.sh
+++ b/init.sh
@@ -530,6 +530,9 @@ function init_hal_power()
 		e-tab*Pro)
 			SLEEP_STATE=force
 			;;
+		*TAIFAElimuTab*)
+			set_property sleep.earlysuspend 1
+			;;
 		*)
 			;;
 	esac
-- 
2.34.1

