From 7bade045a7ce8b26147436065c0b8d4c1138da23 Mon Sep 17 00:00:00 2001
From: electrikjesus <electrikjesus@gmail.com>
Date: Tue, 2 Jan 2024 19:45:50 -0500
Subject: [PATCH 16/43] Enable desktop mode on external displays when PC_MODE
 is enabled

---
 init.sh | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/init.sh b/init.sh
index c903286..9d47ed8 100644
--- a/init.sh
+++ b/init.sh
@@ -924,6 +924,10 @@ function set_custom_package_perms()
 		# allow displaying over other apps if in Go mode
 		settings put system alert_window_bypass_low_ram 1
 
+		# Enable desktop mode on external display (required for MultiDisplay Input)
+		settings put global force_desktop_mode_on_external_displays 1
+		settings put global force_allow_on_external 1
+
 		# Only if PC_MODE is 1
 		if [ $PC_MODE -eq 1 ]; then
 			
-- 
2.34.1

