From 8e98fd48a23b5729751b1b17b2fdeb3f155da97d Mon Sep 17 00:00:00 2001
From: electrikjesus <electrikjesus@gmail.com>
Date: Wed, 10 Jan 2024 20:20:50 -0500
Subject: [PATCH 19/43] pcmode: use HIDE_NAVBAR_WINDOW to override hiding
 navbar window

---
 init.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/init.sh b/init.sh
index 1692732..bc7dda3 100644
--- a/init.sh
+++ b/init.sh
@@ -985,7 +985,7 @@ function set_custom_package_perms()
 			fi
 
 			# Disable Launcher3 Taskbar
-			set_property persist.wm.debug.hide_navbar_window 1
+			set_property persist.wm.debug.hide_navbar_window ${HIDE_NAVBAR_WINDOW:-1}
 
 			[ -n "$SET_SMARTDOCK_DEFAULT" ] && pm set-home-activity "cu.axel.smartdock/.activities.LauncherActivity" || pm set-home-activity "com.android.launcher3/.LauncherProvider"
 			
-- 
2.34.1

