From f1278134e3c64b913665a896cd4717b5ef8a10d6 Mon Sep 17 00:00:00 2001
From: electrikjesus <electrikjesus@gmail.com>
Date: Wed, 10 Jan 2024 20:19:58 -0500
Subject: [PATCH 056/105] Make MultiWindow conditional

---
 device.mk | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/device.mk b/device.mk
index ae3ab60..47c1e6f 100644
--- a/device.mk
+++ b/device.mk
@@ -342,10 +342,13 @@ $(call inherit-product-if-exists, vendor/boringdroid/boringdroid.mk)
 # vendor customization layer
 $(call inherit-product-if-exists, vendor/bass/branding.mk)
 
+
+ifneq ($(or $(USE_BLISS_KIOSK_LAUNCHER), $(USE_BLISS_RESTRICTED_LAUNCHER)),true)
 # Enable MultiWindow
 PRODUCT_PROPERTY_OVERRIDES += \
-    persist.sys.debug.multi_window=true
+    persist.sys.debug.multi_window=true \
     persist.sys.debug.desktop_mode=true
+endif
 
 # DRM service opt-in
 PRODUCT_VENDOR_PROPERTIES += drm.service.enabled=true
-- 
2.34.1

