From f6766a8e6c35e6a643984e7da207a46b34ae4f36 Mon Sep 17 00:00:00 2001
From: electrikjesus <electrikjesus@gmail.com>
Date: Thu, 11 Jan 2024 18:11:55 -0500
Subject: [PATCH 058/105] Remove hide_navbar_window from configs

---
 init.sh | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/init.sh b/init.sh
index 4b1e77e..4b75257 100644
--- a/init.sh
+++ b/init.sh
@@ -823,7 +823,6 @@ function set_custom_package_perms()
 			chmod 775 /data/misc/rlconfig
 			chmod 664 /data/misc/rlconfig/admin
 		fi
-		set_property persist.wm.debug.hide_navbar_window 1
 				
 		pm grant com.bliss.restrictedlauncher android.permission.SYSTEM_ALERT_WINDOW
 		pm set-home-activity "com.bliss.restrictedlauncher/.activities.LauncherActivity"
@@ -951,9 +950,6 @@ function set_custom_package_perms()
 		    	settings put global development_settings_enabled 1
 			fi
 
-			# Disable Launcher3 Taskbar
-			set_property persist.wm.debug.hide_navbar_window ${HIDE_NAVBAR_WINDOW:-1}
-
 			[ -n "$SET_SMARTDOCK_DEFAULT" ] && pm set-home-activity "cu.axel.smartdock/.activities.LauncherActivity" || pm set-home-activity "com.android.launcher3/.LauncherProvider"
 			
 		fi
-- 
2.34.1

