From 4d4373e182f5b47321eb01e3f659b43e49d9a7fb Mon Sep 17 00:00:00 2001
From: electrikjesus <electrikjesus@gmail.com>
Date: Thu, 18 Jan 2024 13:03:51 -0500
Subject: [PATCH 063/105] Add admin support for
 com.bliss.restrictedlauncher.pro

---
 init.sh | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/init.sh b/init.sh
index 08ed428..2fae0db 100644
--- a/init.sh
+++ b/init.sh
@@ -823,6 +823,24 @@ function set_custom_package_perms()
 		am start -a android.intent.action.MAIN -c android.intent.category.HOME
 	fi
 
+	# BlissRestrictedLauncherPro
+	exists_restlauncherpro=$(pm list packages com.bliss.restrictedlauncher.pro | grep -c com.bliss.restrictedlauncher.pro)
+	if [ $exists_restlauncherpro -eq 1 ]; then
+		if [ ! -f /data/misc/rlpconfig/admin ]; then
+			# set device admin
+			dpm set-device-owner com.bliss.restrictedlauncher.pro/com.bliss.restrictedlauncher.DeviceAdmin
+			mkdir -p /data/misc/rlconfig
+			touch /data/misc/rlconfig/admin
+			chown 1000.1000 /data/misc/rlconfig /data/misc/rlconfig/*
+			chmod 775 /data/misc/rlconfig
+			chmod 664 /data/misc/rlconfig/admin
+		fi
+				
+		pm grant com.bliss.restrictedlauncher.pro android.permission.SYSTEM_ALERT_WINDOW
+		pm set-home-activity "com.bliss.restrictedlauncher.pro/com.bliss.restrictedlauncher.activities.LauncherActivity"
+		am start -a android.intent.action.MAIN -c android.intent.category.HOME
+	fi
+
 	# Game-Mode Launcher
 	exists_molla=$(pm list packages com.sinu.molla | grep -c com.sinu.molla)
 	if [ $exists_molla -eq 1 ]; then
-- 
2.34.1

