From cc5913fc158f7e1ba1ccd5c3e6ba8372beda76c0 Mon Sep 17 00:00:00 2001
From: electrikjesus <electrikjesus@gmail.com>
Date: Fri, 23 Feb 2024 09:18:45 -0500
Subject: [PATCH 089/105] Use fuse passthrough

---
 device.mk | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/device.mk b/device.mk
index e4b13d4..4b47535 100644
--- a/device.mk
+++ b/device.mk
@@ -103,6 +103,10 @@ endif
 PRODUCT_PROPERTY_OVERRIDES += \
     ro.hw_timeout_multiplier=3
 
+# Use FUSE passthrough
+PRODUCT_PRODUCT_PROPERTIES += \
+    persist.sys.fuse.passthrough.enable=true
+
 PRODUCT_COPY_FILES := \
     $(if $(wildcard $(PRODUCT_DIR)init.rc),$(PRODUCT_DIR)init.rc:root/init.rc) \
     $(if $(wildcard $(PRODUCT_DIR)init.sh),$(PRODUCT_DIR),$(LOCAL_PATH)/)init.sh:system/etc/init.sh \
@@ -222,7 +226,7 @@ $(foreach f,$(wildcard $(LOCAL_PATH)/permissions_product/*.xml),\
     $(eval PRODUCT_COPY_FILES += $(f):$(TARGET_COPY_OUT_PRODUCT)/etc/permissions/$(notdir $f)))
 
 # Get emulated storage settings
-#$(call inherit-product, $(SRC_TARGET_DIR)/product/emulated_storage.mk)
+$(call inherit-product, $(SRC_TARGET_DIR)/product/emulated_storage.mk)
 
 # Get Android 8.0 HIDL HALs
 $(call inherit-product,$(LOCAL_PATH)/treble.mk)
-- 
2.34.1

