From ecccd4b8283be4d557b3b033c3997ed8d7a9c55f Mon Sep 17 00:00:00 2001
From: electrikjesus <electrikjesus@gmail.com>
Date: Tue, 27 Feb 2024 19:43:40 -0500
Subject: [PATCH 097/105] init: sleep.state defaults to freeze instead of
 s2idle

---
 init.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/init.sh b/init.sh
index 65799f6..6751e15 100644
--- a/init.sh
+++ b/init.sh
@@ -569,7 +569,7 @@ function init_hal_power()
 	if [[ "$mem_sleep_default" == *deep* ]]; then
 		setprop sleep.state ${SLEEP_STATE:-mem}
 	else
-		setprop sleep.state ${SLEEP_STATE:-s2idle}
+		setprop sleep.state ${SLEEP_STATE:-freeze}
 	fi
 
 	current_sleep_state=$(getprop sleep.state)
-- 
2.34.1

